modules/_node-scripts/preflight/checkMainEntryPoints.mjs

```diff
@@ -0,0 +1,72 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+import {SRC_PATH} from '../util/constants.mjs';
+
+export async function checkMainEntryPoints(packageJSONFiles) {
+	const errors = [];
+
+	packageJSONFiles.forEach((pkg) => {
+		const bad = (message) => errors.push(`${pkg.path}: ${message}`);
+
+		try {
+			const {main} = pkg.json;
+
+			const moduleDir = path.join(pkg.path, '..');
+
+			// Check for main entry point
+
+			if (!main) {
+				const indexExists = [
+					'index.js',
+					'index.es.js',
+					'index.ts',
+					'index.tsx',
+				].find(
+					(file) =>
+						fs.existsSync(path.join(moduleDir, SRC_PATH, file)) ||
```

Comment:
Should these use `fileExists` util?

Reply:
I haven't changed all sync stuff to async in this PR because it was already big enough, but we should definitely give that some love in a followup.